src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -22,21 +29,71 @@ internal class VsCodeWindowManager : IVsCodeWindowManager, IVsCodeWindowEvents
             private readonly TLanguageService _languageService;
             private readonly IVsCodeWindow _codeWindow;
             private readonly ComEventSink _sink;
-            private readonly IOptionService _optionService;
+            private readonly IThreadingContext _threadingContext;
+            private readonly IAsynchronousOperationListener _asynchronousOperationListener;
 
-            private INavigationBarController _navigationBarController;
-            private IVsDropdownBarClient _dropdownBarClient;
+            private INavigationBarController? _navigationBarController;
+            private IVsDropdownBarClient? _dropdownBarClient;
+            private IOptionService? _optionService;
+            private WorkspaceRegistration? _workspaceRegistration;
```

Comment:
You _may_ want to add a comment that this is null if we either haven't gotten AddAdornments or already got RemoveAdornments called, and it's used to flag that.